### PR TITLE
limit renaming colony name to 25 characters.

### LINKED ISSUE #204: Limiting the colony name to 25 characters
We should set a limit of 25 chracters for colony names.
This would be useful to prevent bugs from happening.

### PR DESCRIPTION
**Pull request by nojo121**
Closes #204 

# Changes proposed in this pull request:
-limit renaming colony name to 25 characters.

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public TownHallRenameMessage() {}
     public TownHallRenameMessage(@NotNull ColonyView colony, String name)
     {
         this.colonyId = colony.getID();
-        this.name = name;
+        this.name = (name.length() <= 25)? name : name.substring(0, 24);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: MINOR Assign this magic number 25 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java [hunk 2]
```diff
@@ -53,6 +55,7 @@ public IMessage onMessage(@NotNull TownHallRenameMessage message, MessageContext
 
         if (colony != null)
         {
+            this.name = (name.length() <= 25)? name : name.substring(0, 24);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: MINOR Assign this magic number 25 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java (hunk 2 above)
Comment (Raycoms) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: Yeah please assign them to some variable explaining it.

Comment (Raycoms) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: Ty

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java (hunk 2 above)
Comment (Kostronor) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: You still have to replace them on this line ;-)

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java [hunk 3]
```diff
@@ -16,6 +16,8 @@
 {
     private int    colonyId;
     private String name;
+    private static final int MAX_NAME_LENGTH = 25;
+    private static final int SUBSTRING_LENGTH = 24;
```
Comment (Kostronor) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: Suggestion: use Maxnamelength -  1

File: src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java (hunk 2 above)
Comment (nojo121) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: nope

Comment (nojo121) [inline src/main/java/com/minecolonies/network/messages/TownHallRenameMessage.java]: it's just a sonar bug

**Comment by Raycoms** [comment]
:shipit:

**Pull request merged**